kf/bpf.go

```diff
@@ -422,8 +390,9 @@ func (b *BPF) Start(ifaceName, direction string, chain bool) error {
 		}
 	}
 
-	// Fetch when prev program map is updated
-	if len(b.PrevMapNamePath) > 0 {
+	// Fetch when prev program map is updated only when loaded using user program
+	if len(b.PrevMapNamePath) > 0 && b.ProgMapCollection == nil {
```

Comment:
If b.ProgMapCollection is defined, can't we just run whatever operations and checks we need to do for that condition (l3afd managed ebpf) at the top and return?  We should think about restructuring this.

Reply:
The check can only be performed once the user program is invoked.